# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../fusionsound/build-fix.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From: Ville Syrjala <syrjala@sci.fi>
Date: Tue, 21 Oct 2008 13:17:46 +0000 (+0300)
Subject: Fix build with LDFLAGS="-Wl,-no-undefined".
X-Git-Url: http://git.directfb.org/?p=core%2FFusionSound.git;a=commitdiff_plain;h=215457b66c687a64af1bd7e27f5df28bc9c51853

Fix build with LDFLAGS="-Wl,-no-undefined".
---

diff --git a/configure.in b/configure.in
index e827300..322f0d6 100644
--- a/configure.in
+++ b/configure.in
@@ -216,6 +216,8 @@ DIRECT_REQUIRED_VERSION=1.1.0
 
 AC_MSG_CHECKING(for libdirect)
 if $PKG_CONFIG --atleast-version $DIRECT_REQUIRED_VERSION direct ; then
+  DIRECT_CFLAGS=`$PKG_CONFIG --cflags direct`
+  DIRECT_LIBS=`$PKG_CONFIG --libs direct`
   MODULEDIR=$libdir/`$PKG_CONFIG --variable=moduledirname direct`
   AC_MSG_RESULT([found (moduledir = $MODULEDIR)])
 else
@@ -560,6 +562,8 @@ ACLOCAL="$ACLOCAL $ACLOCAL_FLAGS"
 
 AM_CONDITIONAL(BUILD_STATIC, test "$enable_static" = "yes")
 
+AC_SUBST(DIRECT_CFLAGS)
+AC_SUBST(DIRECT_LIBS)
 AC_SUBST(FUSION_CFLAGS)
 AC_SUBST(FUSION_LIBS)
 AC_SUBST(LITE_CFLAGS)
diff --git a/drivers/Makefile.am b/drivers/Makefile.am
index 92b919e..2c71515 100644
--- a/drivers/Makefile.am
+++ b/drivers/Makefile.am
@@ -28,21 +28,21 @@ drivers_LTLIBRARIES = \
 
 libfusionsound_oss_la_SOURCES = oss.c
 
-libfusionsound_oss_la_LIBADD = ../src/libfusionsound.la
+libfusionsound_oss_la_LIBADD = $(DIRECT_LIBS) ../src/libfusionsound.la
 
 libfusionsound_oss_la_LDFLAGS = -export-dynamic -avoid-version -module
 
 
 libfusionsound_alsa_la_SOURCES = alsa.c
 
-libfusionsound_alsa_la_LIBADD = ../src/libfusionsound.la $(ALSA_LIBS)
+libfusionsound_alsa_la_LIBADD = $(DIRECT_LIBS) ../src/libfusionsound.la $(ALSA_LIBS)
 
 libfusionsound_alsa_la_LDFLAGS = -export-dynamic -avoid-version -module
 
 
 libfusionsound_wave_la_SOURCES = wave.c
 
-libfusionsound_wave_la_LIBADD = ../src/libfusionsound.la
+libfusionsound_wave_la_LIBADD = $(DIRECT_LIBS) ../src/libfusionsound.la
 
 libfusionsound_wave_la_LDFLAGS = -export-dynamic -avoid-version -module
 
diff --git a/interfaces/IFusionSoundMusicProvider/Makefile.am b/interfaces/IFusionSoundMusicProvider/Makefile.am
index 55da3cc..901c340 100644
--- a/interfaces/IFusionSoundMusicProvider/Makefile.am
+++ b/interfaces/IFusionSoundMusicProvider/Makefile.am
@@ -53,47 +53,49 @@ ifusionsoundmusicprovider_LTLIBRARIES =	\
 
 libifusionsoundmusicprovider_timidity_la_SOURCES = ifusionsoundmusicprovider_timidity.c
 
-libifusionsoundmusicprovider_timidity_la_LIBADD = $(TIMIDITY_LIBS)
+libifusionsoundmusicprovider_timidity_la_LIBADD = $(DIRECT_LIBS)  $(TIMIDITY_LIBS)
 
 libifusionsoundmusicprovider_timidity_la_LDFLAGS = -avoid-version -module
 
 
 libifusionsoundmusicprovider_wave_la_SOURCES = ifusionsoundmusicprovider_wave.c
 
-libifusionsoundmusicprovider_wave_la_LIBADD =
+libifusionsoundmusicprovider_wave_la_LIBADD = $(DIRECT_LIBS) $(top_builddir)/src/libfusionsound.la
 
 libifusionsoundmusicprovider_wave_la_LDFLAGS = -avoid-version -module
 
 
 libifusionsoundmusicprovider_vorbis_la_SOURCES = ifusionsoundmusicprovider_vorbis.c
 
-libifusionsoundmusicprovider_vorbis_la_LIBADD = $(VORBISFILE_LIBS)
+libifusionsoundmusicprovider_vorbis_la_LIBADD = $(DIRECT_LIBS) $(top_builddir)/src/libfusionsound.la $(VORBISFILE_LIBS)
 
 libifusionsoundmusicprovider_vorbis_la_LDFLAGS = -avoid-version -module
 
 
 libifusionsoundmusicprovider_mad_la_SOURCES = ifusionsoundmusicprovider_mad.c
 
-libifusionsoundmusicprovider_mad_la_LIBADD = $(MAD_LIBS) -lm
+libifusionsoundmusicprovider_mad_la_LIBADD = $(DIRECT_LIBS) $(MAD_LIBS) -lm
 
 libifusionsoundmusicprovider_mad_la_LDFLAGS = -avoid-version -module
 
 
 libifusionsoundmusicprovider_cdda_la_SOURCES = ifusionsoundmusicprovider_cdda.c
 
-libifusionsoundmusicprovider_cdda_la_LIBADD = $(CDDB_LIBS)
+libifusionsoundmusicprovider_cdda_la_LIBADD = $(DIRECT_LIBS) $(CDDB_LIBS)
 
 libifusionsoundmusicprovider_cdda_la_LDFLAGS = -avoid-version -module
 
 
 libifusionsoundmusicprovider_playlist_la_SOURCES = ifusionsoundmusicprovider_playlist.c
 
+libifusionsoundmusicprovider_playlist_la_LIBADD = $(DIRECT_LIBS) $(top_builddir)/src/libfusionsound.la
+
 libifusionsoundmusicprovider_playlist_la_LDFLAGS = -avoid-version -module
 
 
 libifusionsoundmusicprovider_ffmpeg_la_SOURCES = ifusionsoundmusicprovider_ffmpeg.c
 
-libifusionsoundmusicprovider_ffmpeg_la_LIBADD = $(FFMPEG_LIBS)
+libifusionsoundmusicprovider_ffmpeg_la_LIBADD = $(DIRECT_LIBS) $(top_builddir)/src/libfusionsound.la $(FFMPEG_LIBS)
 
 libifusionsoundmusicprovider_ffmpeg_la_LDFLAGS = -avoid-version -module
 
